cmd/sops/main.go

```diff
@@ -733,12 +737,16 @@ func main() {
 				return common.NewExitError("Error: cannot operate on non-existent file", codes.NoFileSpecified)
 			}
 		}
+		fileNameOverride := c.String("filename-override")
+		if fileNameOverride == "" {
+			fileNameOverride = fileName
+		}
```

Comment:
Curious why choose `fileNameOverride` vs modify `fileName`?

```go
		fileNameOverride := c.String("filename-override")
		if fileNameOverride != "" {
			fileName = fileNameOverride
		}
```

Related do we need to update this `fileName` reference on lines 976-982?

```go
		// We open the file *after* the operations on the tree have been
		// executed to avoid truncating it when there's errors
		if c.Bool("in-place") || isEditMode || c.String("set") != "" {
			file, err := os.Create(fileName)
			if err != nil {
				return common.NewExitError(fmt.Sprintf("Could not open in-place file for writing: %s", err), codes.CouldNotWriteOutputFile)
			}
```

Reply:
> Curious why choose `fileNameOverride` vs modify `fileName`?

Because we still want to read from / write to `fileName`, to we need the original value for these operations.

> Related do we need to update this `fileName` reference on lines 976-982?

No, because `fileName` is the file we actually read from (and write to, if `--output` isn't used).